text/0000-export-to-oci.md

```diff
@@ -0,0 +1,715 @@
+
+  Some thoughts about this approaches
+
+    - **Process Management:** Platforms must now manage a parallel process (registry in the daemon). This would entail ensuring that the registry is started and cleaned up appropriately.
+    - **Networking:** There are additional network complications in order to route images to the ephemeral registry. For example, [network drivers](https://docs.docker.com/network/#network-drivers), [proxy](https://docs.docker.com/desktop/networking/#httphttps-proxy-support) and [DNS configuration](https://docs.docker.com/config/containers/container-networking/#dns-services), [host name resolution](https://docs.docker.com/desktop/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host), and [TLS certificates](https://betterprogramming.pub/deploy-a-docker-registry-using-tls-and-htpasswd-56dd57a1215a) to name a few.
+
+- Why is this proposal the best? [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format is a standard from which other tools can create a [OCI Runtime Specification bundle](https://github.com/opencontainers/runtime-spec/blob/v1.0.0/bundle.md) exporting to this format enables Platforms to implement any feature in the top of this format, for example, exporting to [containerd](https://containerd.io) has been [requested](https://github.com/buildpacks/lifecycle/issues/829) by the community and it could be implemented if we can get the application image exported in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format.
+- What is the impact of not doing this? Probably will never remove the Daemon support in the Lifecycle
+
+# Prior Art
+[prior-art]: #prior-art
+
+- Discussion around removing the Daemon support [RFC](https://github.com/buildpacks/rfcs/blob/jjbustamante/feature/deprecate-daemon/text/0000-deprecate-daemon.md)
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- What parts of the design do you expect to be resolved before this gets merged?
+  - Tools like [umoci](https://umo.ci/) used to create a runtime bundle from an image in OCI layout format, requires the [annotation](https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys) `org.opencontainers.image.ref.name` to be present. Also tools like [skopeo](https://github.com/containers/skopeo) when an image is `copy` in oci format the annotation is included.
+  We are not adding the annotation as part of the Buildpacks Specification, but in this case this could make our output incompatible with some other tooling.
+  - Depending on the previous question, the rules for mapping an image reference into a path could change, also validations that the Lifecycle could do to verify the `org.opencontainers.image.ref.name` against the value provided in the image reference.
```

Comment:
Can you explain a bit more about what problems would arise? Do we have to validate anything at first?

Reply:
Yes, right now the rules to map the `image.ref` to a path are:

- `<registry>/<repo>/<image>:<tag>` is converted into:
  -  `<layout-dir>/<registry>/<repo>/<image>/<tag>`

- `<registry>/<repo>/<image>@<algorithm>:<digest>` is converted into:
  - `<layout-dir>/<registry>/<repo>/<image>/<algorithm>/<digest>`

My main concern was if we handle the `org.opencontainers.image.ref.name` annotation then the first case could be converted into `<layout-dir>/<registry>/<repo>/<image>` and lifecycle could validate if the `<tag>` provided matches the `org.opencontainers.image.ref.name` annotation of the image at path `<layout-dir>/<registry>/<repo>/<image>` BUT thinking a little bit more, I think the validation can exists, but the mappings rules are fine, because if I want to use `ubuntu:bionic` or `ubuntu:jammy` if I do not include the bionic or jammy tag as part of the path, then it could be complicated to resolve the image. 

Don't worry about this statement! I think it is not going to change the mapping rules. 😄